modules/tinymce/src/plugins/lists/main/ts/core/Delete.ts

```diff
@@ -61,8 +61,12 @@ const hasOnlyOneBlockChild = (dom: DOMUtils, elm: Element): boolean => {
   return childNodes.length === 1 && !NodeType.isListNode(childNodes[0]) && dom.isBlock(childNodes[0]);
 };
 
+const isUnwrappable = (elm: HTMLElement): boolean =>
+  ContentEditable.isEditable(SugarElement.fromDom(elm))
+    && !Arr.contains([ 'DETAILS' ], elm.nodeName);
+
 const unwrapSingleBlockChild = (dom: DOMUtils, elm: Element): void => {
-  if (hasOnlyOneBlockChild(dom, elm)) {
+  if (hasOnlyOneBlockChild(dom, elm) && isUnwrappable(elm.firstChild as HTMLElement)) {
     dom.remove(elm.firstChild as HTMLElement, true);
   }
 };
```

Comment:
Maybe we should avoid dangerous type casts and do something like this.

```
const isUnwrappable = (node: Node | null): node is HTMLElement =>
  Optional.from(node)
    .map(SugarElement.fromDom)
    .filter(SugarNode.isHTMLElement)
    .exists((el) => ContentEditable.isEditable(el) && Arr.contains([ 'details' ], SugarNode.name(el)));

const unwrapSingleBlockChild = (dom: DOMUtils, elm: Element): void => {
  if (hasOnlyOneBlockChild(dom, elm) && isUnwrappable(elm.firstChild)) {
    dom.remove(elm.firstChild, true);
  }
};

```

Reply:
Thanks, that looks much better